src/Cake.Core/Scripting/Processors/Loading/FileLoadDirectiveProvider.cs

```diff
@@ -54,14 +54,16 @@ public void Load(IScriptAnalyzerContext context, LoadReference reference)
             var current = context.Current.Path.GetDirectory();
             path = path.MakeAbsolute(current);
 
+            var expectedExtension = path.HasExtension ? path.GetExtension() : ".cake";
```

Comment:
This check handles local file specifications authored prior to https://github.com/cake-build/cake/commit/0488e2eff24f86240684216fd2e34a470745b721 and simple glob patterns (eg: *.cs, scripts/*.cs). It doesn't handle more complex globs such as *.{cs,kake}.

My first thought was to use a FileExists check but this class doesn't have access to IEnvironment or any other mechanism for performing that check.